org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/util/JavaProjectSetupUtil.java

```diff
@@ -81,32 +81,29 @@ public TextFile(String path, String content) {
 	 * creates a JarInputStream containing the passed text files. Each Pair<String
 	 */
 	public static InputStream jarInputStream(TextFile... files) {
-		try {
-			ByteArrayOutputStream out2 = new ByteArrayOutputStream();
-			JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2));
+		try (ByteArrayOutputStream out2 = new ByteArrayOutputStream();
+				JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2))) {
 			for (TextFile textFile : files) {
 				JarEntry je = new JarEntry(textFile.path);
 				jo.putNextEntry(je);
 				byte[] bytes = textFile.content.getBytes();
 				jo.write(bytes, 0, bytes.length);
 			}
-			jo.close();
+			jo.flush();
 			return new ByteArrayInputStream(out2.toByteArray());
 		} catch (IOException e) {
 			throw new WrappedException(e);
 		}
 	}
 	
 	public static InputStream jarInputStream(@SuppressWarnings("unchecked") Pair<String, InputStream> ...entries) {
-		try {
-			ByteArrayOutputStream out2 = new ByteArrayOutputStream();
-			JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2));
+		try (ByteArrayOutputStream out2 = new ByteArrayOutputStream(); JarOutputStream jo = new JarOutputStream(new BufferedOutputStream(out2))) {
 			for (Pair<String, InputStream> entry : entries) {
 				JarEntry je = new JarEntry(entry.getKey());
 				jo.putNextEntry(je);
 				ByteStreams.copy(entry.getValue(), jo);
 			}
-			jo.close();
+			jo.flush();
```

Comment:
We do need the close on the jo because the deflater needs to write the finishing bytes.

Reply:
Thanks for the carefull review :)